File: examples/spring-boot-maven-plugin/src/main/java/community/flock/wirespec/examples/app/exception/AppException.java
package community.flock.wirespec.examples.app.exception;

public sealed class AppException extends RuntimeException permits Conflict, NotFound, CallInterrupted {
    public AppException(String message) {
        super(message);
    }

    public AppException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: examples/spring-boot-gradle-plugin/src/main/kotlin/exception/AppException.kt
package community.flock.wirespec.examples.app.exception

import community.flock.wirespec.examples.app.common.invoke
import community.flock.wirespec.examples.app.todo.Todo

sealed class AppException(override val message: String) : RuntimeException(message)

sealed class TodoException(message: String) : AppException(message)

class TodoIdNotValidException(invalidId: String) : TodoException("TodoId: $invalidId, not valid")

class TodoNotFoundException(id: Todo.Id) : TodoException("Todo with id: ${id()}, not found")
